code/controllers/subsystem/machinery.dm

```diff
@@ -26,16 +29,22 @@ var/list/machines = list()
 
 // This is to allow the near identical fast machinery process to use it.
 /datum/subsystem/machinery/proc/get_currenrun()
-	return machines.Copy()
+	currentrun_index = machines.len
+	currentrun.len = currentrun_index
+	if (currentrun_index)
+		for(c in 1 to currentrun_index)
+		currentrun[c] = machines[c]
```

Comment:
This seems to be indented incorrectly, which could be why you're seeing a speed-up, since this line only runs once instead of current_index times.